gh-130474: Create and implement flaky resource (#130489)
python/cpython: Misc/NEWS.d/next/Tests/2025-02-23-21-20-52.gh-issue-130474.dEckCL.rst

> Add the ``flaky`` resource for flagging tests with flaky outcomes using the :func:`test.support.required_resource` decorator.

Reviewer: ```suggestion
Add the ``flaky`` resource for flagging tests with flaky outcomes using the :func:`!test.support.required_resource` decorator.
```

> Add the ``flaky`` resource for flagging tests with flaky outcomes using the :func:`!test.support.required_resource` decorator.

Reviewer 2: ```suggestion
Add the ``flaky`` resource for flagging tests with unreliable outcomes
using the :py:deco:`!test.support.required_resource` decorator.
```

> Created flaky resource for flagging tests with flaky outcomes using the support.required_resource() decorator. Flagged some initial tests based on-inline comments. This resource can be used to automate re-running certain tests in CI.

Reviewer: ```suggestion
Add the ``flaky`` resource for flagging tests with flaky outcomes using the :func:`test.support.required_resource` decorator.
```
